### ISSUE TITLE
Cannot overwrite file type specific settings unless exactly specify that file type

### ISSUE THREAD

**Issue by adusamr** [microsoft/vscode#233505]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.95.2 (user setup)
- OS Version: Windows_NT x64 10.0.22631

defaultSettings.json is like this:
```json
{
	"editor.unicodeHighlight.ambiguousCharacters": true,
	"editor.unicodeHighlight.invisibleCharacters": true,
	"[markdown]":  {
		"editor.unicodeHighlight.ambiguousCharacters": false,
		"editor.unicodeHighlight.invisibleCharacters": false,
	}
}
```
In user's settings.json,
```json
{
	"editor.unicodeHighlight.ambiguousCharacters": true,
	"editor.unicodeHighlight.invisibleCharacters": true,
}
```
or
```json
{
	"[markdown][plaintext]":  {
		"editor.unicodeHighlight.ambiguousCharacters": true,
		"editor.unicodeHighlight.invisibleCharacters": true,
	}
}
```
do not overwrite these settings in markdown.
Only
```json
{
	"[markdown]":  {
		"editor.unicodeHighlight.ambiguousCharacters": true,
		"editor.unicodeHighlight.invisibleCharacters": true,
	}
}
```
can overwrite these settings.

I don't know whether this happens for other settings.
I found this when testing https://github.com/microsoft/vscode/issues/233503#issue-2646626769

**Comment by sandy081** [comment]
@adusamr It is not clear to me from the issue description what is expected and what is actually happening. Can you please update the issue and let us know what you expect and what you are observing. Thanks.

**Comment by adusamr** [comment]
I read https://code.visualstudio.com/docs/getstarted/settings and now I understand a part of the issue was documented behavior.

Then the real problem or undocumented behavior is "single language-specific settings is preceding to multiple language-specific settings" even if the single language-specific settings are default settings and  the multiple language-specific settings are user settings.

![Image](https://github.com/user-attachments/assets/4884bac2-2fb6-4495-a68c-63d2ae52f931)
The defaultSettings.json is:
```json
{
	"editor.fontSize": 14,
	"editor.wordWrap": "off",
	"[markdown]":  {
		"editor.wordWrap": "on"
	}
}
```
In the image, the right column is user's settings.json.

The middle column is plaintext and its font size is 20 and word wrap is off. The "[markdown][plaintext]" user settings is correctly working.

The left column is markdown.
Its font size is 8, which means the "[markdown]" user settings is preceding to the "[markdown][plaintext]" user settings.
And word wrap is on, which means the "[markdown]" **default** settings is preceding to the "[markdown][plaintext]" user settings.

If you uncomment the `"editor.wordWrap": "off"` line in the "[markdown]" user settings (and reopen markdown.md), now you can see word wrap is off.

**Comment by sandy081** [comment]
Yes, single language specific settings takes precedence over multiple language specific settings. I will move this to our docs repo to make sure it is documented.
